Source/WebCore/platform/PublicSuffix.h

```diff
@@ -25,13 +25,33 @@
 
 #pragma once
 
+#include <wtf/HashMap.h>
+#include <wtf/HashSet.h>
+#include <wtf/text/StringHash.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
-WEBCORE_EXPORT bool isPublicSuffix(StringView domain);
-WEBCORE_EXPORT String topPrivatelyControlledDomain(const String& domain);
-WEBCORE_EXPORT void setTopPrivatelyControlledDomain(const String& domain, const String& topPrivatelyControlledDomain);
-String decodeHostName(const String& domain);
+class PublicSuffixStore {
+public:
+    WEBCORE_EXPORT static PublicSuffixStore& singleton();
+    PublicSuffixStore() = default;
+
+    WEBCORE_EXPORT void enablePublicSuffixCache();
+    WEBCORE_EXPORT void addPublicSuffix(const String& domain);
+
+    WEBCORE_EXPORT bool isPublicSuffix(StringView domain);
```

Comment:
If any of these are driven by spec algorithms, I recommend adding comments with links to the specs where you can, to clarify what exactly the functions intention is.